data-pipeline-ffi/src/error.rs

```diff
@@ -0,0 +1,161 @@
+            Self::NetworkUnknown => write!(f, "Unknown network error"),
+            Self::Serde => write!(f, "Serialization/Deserialization error"),
+            Self::TimedOut => write!(f, "Operation timed out"),
+        }
+    }
+}
+
+/// Stucture that contains error information that `TraceExporter` API can return.
+#[repr(C)]
+#[derive(Debug, PartialEq)]
+pub struct ExporterError {
+    pub code: ExporterErrorCode,
+    pub msg: *mut c_char,
+}
+
+impl ExporterError {
+    pub fn new(code: ExporterErrorCode, msg: &str) -> Self {
+        Self {
+            code,
+            msg: CString::new(msg).unwrap_or_default().into_raw(),
+        }
+    }
+}
+
+impl From<TraceExporterError> for ExporterError {
+    // add code here
+    fn from(value: TraceExporterError) -> Self {
+        let code = match &value {
+            TraceExporterError::Builder(e) => match e {
+                BuilderErrorKind::InvalidUri => ExporterErrorCode::InvalidUrl,
+            },
+            TraceExporterError::Deserialization(_) => ExporterErrorCode::Serde,
+            TraceExporterError::Io(e) => match e.kind() {
+                IoErrorKind::InvalidData => ExporterErrorCode::InvalidData,
+                IoErrorKind::InvalidInput => ExporterErrorCode::InvalidInput,
+                IoErrorKind::ConnectionReset => ExporterErrorCode::ConnectionReset,
+                IoErrorKind::ConnectionAborted => ExporterErrorCode::ConnectionAborted,
+                IoErrorKind::ConnectionRefused => ExporterErrorCode::ConnectionRefused,
+                IoErrorKind::TimedOut => ExporterErrorCode::TimedOut,
+                IoErrorKind::AddrInUse => ExporterErrorCode::AddressInUse,
+                _ => ExporterErrorCode::IoError,
+            },
+            TraceExporterError::Network(e) => match e.kind() {
+                NetworkErrorKind::Body => ExporterErrorCode::HttpBodyFormat,
+                NetworkErrorKind::Canceled => ExporterErrorCode::ConnectionAborted,
+                NetworkErrorKind::ConnectionClosed => ExporterErrorCode::ConnectionReset,
+                NetworkErrorKind::MessageTooLarge => ExporterErrorCode::HttpBodyTooLong,
+                NetworkErrorKind::Parse => ExporterErrorCode::HttpParse,
+                NetworkErrorKind::TimedOut => ExporterErrorCode::TimedOut,
+                NetworkErrorKind::Unknown => ExporterErrorCode::NetworkUnknown,
+                NetworkErrorKind::WrongStatus => ExporterErrorCode::HttpWrongStatus,
+            },
+            TraceExporterError::Request(e) => {
+                let status: u16 = e.status().into();
+                if (400..500).contains(&status) {
+                    ExporterErrorCode::HttpClient
+                } else {
+                    ExporterErrorCode::HttpServer
+                }
+            }
+            TraceExporterError::Serde(_) => ExporterErrorCode::Serde,
+        };
+        ExporterError::new(code, &value.to_string())
+    }
+}
+
+/// Free
```

Comment:
Does this warrant a more detailed docstring comment? Or should it be self-explanatory?